### PR TITLE
Day 1 Summary

### PR DESCRIPTION
**Pull request by YonLiud**
Usage of the API:
Other smaller projects will connect to this API instead of messing with the database, the API will provide easy access to the applications to read information about other contacts

Done the Flask application so other devices can send an HTTP request to the pages,
Done an API key generator with the ability to verify it
Done a working SQLite database for contacts

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        return jsonify({
+            'error': "API key is not valid"
+        })
+    try:
+        for account in accounts:
+            if int(account["contact_id"]) == int(id): return jsonify(account)
+        else:
+            return jsonify({
+                'error': 'Account not found'
+            })
```
Comment (katzio) [inline app.py:95]: no idea what is happening here :/

**Review by katzio — COMMENTED** [review]
please read and practice 
https://www.digitalocean.com/community/tutorials/build-a-crud-web-app-with-python-and-flask-part-one

File: app.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    cursor = conn.cursor()
+    desc =  cursor.execute("pragma table_info('contacts')").fetchall()
+    for i in range(1, cursor.execute('SELECT COUNT(*) from contacts').fetchall()[0][0]+1):
+        account = {}
+        name = ""
+        value = []
+        try:
+            for column in desc:
+                name = column[1]
+                value = cursor.execute('SELECT ' + name + ' FROM contacts WHERE contact_id='+str(i)).fetchall()
```
[the remark this answers is not in the document]
Comment (YonLiud) [reply app.py:19]: how can you inject anything?? that's automated by simply creating a dictionary for each value in the DB

File: app.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    return cursor.execute('SELECT key FROM keys').fetchall()
+
+def validatekey(key):
+    keys_conn = sqlite3.connect('api_keys.db')
+    valid_keys = get_api_keys(keys_conn)
+    if not valid_keys:
+        return 0
+    for validate in valid_keys:
+        if key == validate[0]:
+            return 1
```
[the remark this answers is not in the document]
Comment (YonLiud) [reply app.py:42]: that's.... cool actually I didn't know u can do this stuff, will add to next commit

File: app.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    else:
+        return 0
+    keys_conn.close()
+def register_key(name, email):
+    keys_conn = sqlite3.connect('api_keys.db')
+    cursor = keys_conn.cursor()
+    generated_key = secrets.token_urlsafe(16)
+    try:
+        cursor.execute('INSERT INTO keys VALUES(?, ?, ?)', (generated_key, name, email))
```
[the remark this answers is not in the document]
Comment (YonLiud) [reply app.py:51]: hmmmmmmm... will take into consideration for next commits and updates

File: app.py (hunk 1 above)
Comment (YonLiud) [reply app.py:95]: a simple 'for' loop to locate a specific account by a requested ID

File: app.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                if not value:
+                    value = 'void'
+                else:
+                    value = value[0][0]
+                account[name] = value
+            accounts.append(account)
+        except Exception as exc:
+            return jsonify({
+            'error': exc
+            })
```
[the remark this answers is not in the document]
Comment (YonLiud) [reply app.py:29]: no, methods are the satan, * hisses *

File: app.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+def get_api_keys(keys_conn):
+    cursor = keys_conn.cursor()
+    return cursor.execute('SELECT key FROM keys').fetchall()
+
+def validatekey(key):
```
[the remark this answers is not in the document]
Comment (YonLiud) [reply app.py:35]: https://stackoverflow.com/questions/2764017/is-false-0-and-true-1-an-implementation-detail-or-is-it-guaranteed-by-the

File: app.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (YonLiud) [reply app.py:51]: image
 yeah im pretty sure the method I did is the official method SQLite suggested in their python docs

**Pull request merged**